src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -136,5 +138,41 @@ public static FormulaValue CountRows_UO(IRContext irContext, UntypedObjectValue[
 
             return CommonErrors.RuntimeTypeMismatch(irContext);
         }
+
+        public static FormulaValue DateValue_UO(IRContext irContext, UntypedObjectValue[] args)
+        {
+            var impl = args[0].Impl;
+
+            if (impl.Type == FormulaType.String)
+            {
+                var s = impl.GetString();
+                if (DateTime.TryParseExact(s, "yyyy-MM-dd", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime res))
+                {
+                    return new DateValue(irContext, res.Date);
+                }
+
+                return CommonErrors.InvalidDateTimeError(irContext);
+            }
+
+            return CommonErrors.RuntimeTypeMismatch(irContext);
+        }
+
+        public static FormulaValue DateTimeValue_UO(IRContext irContext, UntypedObjectValue[] args)
+        {
+            var impl = args[0].Impl;
+
+            if (impl.Type == FormulaType.String)
+            {
+                var s = impl.GetString();
+                if (DateTime.TryParseExact(s, "yyyy-MM-dd'T'HH:mm:ss.FFFK", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime res))
```

Comment:
maybe add link to documentation on this "magic string"?

Reply:
Done